### ISSUE TITLE
Comment code and add to documentation

### ISSUE THREAD

**Issue by jaredballou** [djfdyuruiry/cnc-td-nco-mod#39]
There's a growing amount of code we've added to the project and there's not a lot of comments or documentation. I'm going to go through my code and add some comments and update the documents, but the new Lua library in particular probably ought to get some more love so that I can know what the heck it's doing :D

**Comment by djfdyuruiry** [comment]
I agree - but if the code is unclear we should refactor it to be more understandable. Although - some of the more c/c++ heavy parts will never get there 🙈 so point taken. Also this should be rolled into the effort to get the lua api to be reflective and auto generate docs
